ecl/hqlcpp/hqlhtcpp.cpp

```diff
@@ -18650,15 +18650,18 @@ void HqlCppTranslator::doBuildNewRegexFindReplace(BuildCtx & ctx, const CHqlBoun
         return;
     }
 
-    IHqlExpression * pattern = expr->queryChild(0);
-    IHqlExpression * search = expr->queryChild(1);
-    ITypeInfo * searchStringType = search->queryType();
-    IHqlExpression * compiled = doBuildRegexCompileInstance(ctx, pattern, searchStringType, !expr->hasAttribute(noCaseAtom));
+    IHqlExpression * patternExpr = expr->queryChild(0);
+    OwnedITypeInfo patternStringType = patternExpr->queryType();
```

Comment:
Note: Changing 'determine the data type of all arguments' from subject to pattern in order to match parser logic.